pkg/storage/repo/storage.go

```diff
@@ -0,0 +1,269 @@
+		if err := os.RemoveAll(dir); err != nil {
+			return fmt.Errorf("unable to remove storage folder: %w", err)
+		}
+		return nil
+	})
+	storageHandler := &StorageProvider{
+		LocalDir:      dir,
+		EstuaryAPIKey: EstuaryAPIKey,
+		IPFSClient:    IPFSapiclient,
+		CloneClient:   c,
+	}
+	log.Debug().Msgf("Repo download driver created with output dir: %s", dir)
+	return storageHandler, nil
+}
+
+func (sp *StorageProvider) IsInstalled(context.Context) (bool, error) {
+	err := checkGitLFS()
+	return err == nil, err
+}
+
+func (sp *StorageProvider) HasStorageLocally(context.Context, model.StorageSpec) (bool, error) {
+	return false, nil
+}
+
+// Could do a HEAD request and check Content-Length, but in some cases that's not guaranteed to be the real end file size
+func (sp *StorageProvider) GetVolumeSize(context.Context, model.StorageSpec) (uint64, error) {
+	return 0, nil
+}
+
+//nolint:gocyclo
+func (sp *StorageProvider) PrepareStorage(ctx context.Context, storageSpec model.StorageSpec) (storage.StorageVolume, error) {
+	_, span := system.GetTracer().Start(ctx, "pkg/storage/repo/repo.PrepareStorage")
+	defer span.End()
+	repoURL := storageSpec.Repo
+	var err error
+	_, err = clone.IsValidGitRepoURL(repoURL)
+	// fmt.Printf("%+v", storageSpec)
+	if err != nil {
+		return storage.StorageVolume{}, err
+	}
+	// # create a tmp directory
+	outputPath, err := os.MkdirTemp(sp.LocalDir, "*")
+	fmt.Printf("Output Path: %s\n", outputPath)
+	if err != nil {
+		return storage.StorageVolume{}, err
+	}
+
+	// err = git.Clone(repoURL, outputPath)
+	// if err != nil {
+	// 	fmt.Println("Failed to clone repository:", err)
+	cmd := exec.Command("git", "clone", repoURL, outputPath)
+	// The `Output` method executes the command and
+	// collects the output, returning its value
+	out, err1 := cmd.Output()
+	if err1 != nil {
+		// if there was any error, print it here
+		fmt.Println("could not run command: ", err)
+	}
+	// otherwise, print the output from running the command
+	fmt.Println("Output: ", string(out))
+	if err != nil {
+		panic(err)
+	}
+
+	if err != nil {
+		return storage.StorageVolume{}, err
+	}
+	// }
+	filepath, err2 := url.Parse(repoURL)
+	if err2 != nil {
+		return storage.StorageVolume{}, err
+	}
+	filename := strings.Split(filepath.Path, ".")[0]
+	targetPath := "/inputs" + filename
+
+	CIDSpec, err := sp.Upload(ctx, outputPath)
+	// If estuary key exists then upload it to estuary
+	if err != nil {
+		return storage.StorageVolume{}, err
+	}
+	CID := CIDSpec.CID
+	// Update the KV store
+	SHA1HASH, _ := UrltoLatestCommitHash(repoURL)
+	spkey := sp.EstuaryAPIKey
+	envkey := os.Getenv("ESTUARY_API_KEY")
+	var key string
+	if spkey != "" && envkey == "" {
+		key = spkey
+	} else if envkey != "" && spkey == "" {
+		key = envkey
+	} else if envkey != "" && spkey != "" {
+		key = spkey
+	} else {
+		key = ""
+	}
+	if key != "" {
+		log1.Println("Pinning to Estuary...")
+		//nolint:govet,noctx
+		err := estuary.PinToIPFSViaEstuary(ctx, key, CID)
```

Comment:
The responsibility of this class is to allow users to use git repos directly in Bacalhau. Pinning repos to Estuary is a separate job to be done. This class should be usable even on nodes that aren't connected to Estuary (and we currently aren't even configuring Estuary on our nodes, so it won't work in prod). 

You need to take the Estuary pinning out and put it in a separate part, I'd suggest part of the job transform discussed above that looks at whether the repo is already on IPFS.

Reply:
> This class should be usable even on nodes that aren't connected to Estuary (and we currently aren't even configuring Estuary on our nodes, so it won't work in prod).

I think it will not execute on node that aren't configuring estuary